realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -504,4 +514,14 @@ private int getTotalGlobalRefCount() {
 
         return totalRefCount;
     }
+
+    /**
+     * If a Realm instance is GCed but `Realm.close()` is not called before, we still want to track the cache for
+     * debugging. Adding them to the list to keep the strong ref of the cache to prevent the cache gets GCed.
+     */
+    void leak() {
+        if (!isLeaked.getAndSet(true)) {
+            leakedCaches.add(this);
```

Comment:
hmm, the key (thread id?) is weak ref
```
        static class Entry extends WeakReference<ThreadLocal> {
            /** The value associated with this ThreadLocal. */
            Object value;

            Entry(ThreadLocal k, Object v) {
                super(k);
                value = v;
            }
        }
```

So the case you show won't happen as long as the thread is not terminated? `leak()` won't be called in that case.